packages/pigeon/lib/dart_generator.dart

```diff
@@ -55,6 +70,32 @@ class DartOptions {
   }
 }
 
+/// Class that manages all Dart code generation.
+class DartGenerator extends Generator<DartOptions> {
+  /// Instantiates a Dart Generator.
+  DartGenerator();
+
+  /// Generates Dart files with specified [DartOptions]
+  @override
+  void generate(
+      DartOptions languageOptions, Root root, StringSink sink, FileType _) {
+    generateDart(languageOptions, root, sink);
+  }
+
+  /// Generates Dart files for testing with specified [DartOptions]
+  void generateTest(DartOptions languageOptions, Root root, StringSink sink) {
```

Comment:
Optional: Why not make this a file type as well so the structure is the same?

Reply:
Just seemed like unneeded  code. I thought about it. If you think that's best, I will add it though.

Reply:
I was considering that, I just didn't think about it when I wrote the code. I'll change it.

Reply:
The issue with this, upon thinking about it more, is that it would generate the test files when generating non test dart code as well. I don't know if that is super important to avoid, but it seems unnecessary.

Reply:
We might be able to avoid that by making the set of types option-driven, but let's not worry about it now and just leave it as you have it; we can always revisit later and it's a minor thing.